lib/syntax_tools/src/erl_syntax.erl

```diff
@@ -6208,10 +6228,60 @@ binary_generator_body(Node) ->
 block_expr(Body) ->
     tree(block_expr, Body).
 
+%% =====================================================================
+%% @doc Creates an abstract block expression. If `Body' is
+%% `[B1, ..., Bn]', and `Clauses' is `[C1, ..., Cn]', 
+%% the result represents "<code>begin
+%% <em>B1</em>, ..., <em>Bn</em> else <em>C1</em>; ...,
+%% <em>Cn</em> end</code>".
+%%
+%%	See `clause' for documentation on `erl_parse' clauses.
+%% @see block_expr_body/1
+
+%% type(Node) = block_expr
+%% data(Node) = #block_expr{body :: Body, clauses :: Clauses}
+%%
+%%     Body = [syntaxTree()]
+%%     Clauses = [syntaxTree()]
+%%
+%% `erl_parse' representation:
+%%
+%% {block, Pos, Body, Clauses}
+%%
+%%    Body = [erl_parse()] \ []
+%%    Clauses = [Clause] \ []
+%%    Clause = {clause, ...}
+
+-spec block_expr([syntaxTree()], [syntaxTree()]) -> syntaxTree().
+
+block_expr(Body, Clauses) ->
+    tree(block_expr, #block_expr{body = Body,
+                                 clauses = Clauses}).
+
 revert_block_expr(Node) ->
-    Pos = get_pos(Node),
-    Body = block_expr_body(Node),
-    {block, Pos, Body}.
+    case unwrap(Node) of
```

Comment:
this bit is really messy because I try to support both forms and that doesn't go super well, but it works